### PR TITLE
Migrate custom BEM to getBEM syntax

### PR DESCRIPTION
**Pull request by anselmbradford**
Bring in updated getBEM name work from DS (see https://github.com/cfpb/design-system/pull/1919)

## Changes

- Update to v1.0.0 of the design system.
- Migrate custom BEM to getBEM syntax. The new syntax separates elements with a double underscore `__` and modifiers with a double dash `--`.
- Adds `selector-class-pattern` setting for BEM to stylelint.

## How to test this PR

1. PR checks should pass and the site should be the same, which is a big ask, but this touches pretty much everywhere 🙀 
2. `yarn build` and `yarn lint` should pass.

### CODE REVIEW

File: cfgov/tccp/jinja2/tccp/landing_page.html [hunk 1]
```diff
@@ -38,7 +38,7 @@
     {% include "v1/includes/molecules/hero.html" with context %}
 {% endwith %}
 
-{% endblock content_intro %}
+{% endblock content__intro %}
```
Comment (contolini) [inline cfgov/tccp/jinja2/tccp/landing_page.html:41]:
```suggestion
{% endblock content_intro %}
```

File: cfgov/tccp/jinja2/tccp/landing_page.html [hunk 2]
```diff
@@ -121,8 +121,8 @@ <h2>
         </form>
     </div>
 
-{% endblock content_main %}
+{% endblock content__main %}
 
 {% block content_sidebar %}
     {%- include "tccp/includes/sidebar.html" -%}
-{% endblock content_sidebar %}
+{% endblock content__sidebar %}
```
Comment (contolini) [inline cfgov/tccp/jinja2/tccp/landing_page.html:128]:
```suggestion
{% endblock content_sidebar %}
```

File: cfgov/teachers_digital_platform/jinja2/teachers_digital_platform/activity_page.html [hunk 3]
```diff
@@ -88,7 +88,7 @@ <h2>Explore related resources</h2>
             <li><a href="https://www.federalreserveeducation.org/" target="_blank" rel="noopener noreferrer">View financial education resources from the Federal Reserve</a></li>
         </ul>
     </div>
-{% endblock content_main %}
+{% endblock content__main %}
```
Comment (contolini) [inline cfgov/teachers_digital_platform/jinja2/teachers_digital_platform/activity_page.html:91]:
```suggestion
{% endblock content_main %}
```

File: cfgov/mega_menu/jinja2/mega_menu/preview.html [hunk 4]
```diff
@@ -57,4 +57,4 @@
 eleifend risus quis finibus. Sed aliquam ullamcorper ipsum non fermentum.
 Vivamus malesuada orci et neque laoreet, id pretium odio dignissim.
 </p>
-{% endblock content_main %}
+{% endblock content__main %}
```
Comment (contolini) [inline cfgov/mega_menu/jinja2/mega_menu/preview.html:60]:
```suggestion
{% endblock content_main %}
```

File: cfgov/regulations3k/jinja2/regulations3k/landing-page.html [hunk 5]
```diff
@@ -34,7 +34,7 @@
     {% for block in page.content -%}
         {{ render_block.render(block, loop.index) }}
     {%- endfor %}
-{% endblock content_main %}
+{% endblock content__main %}
```
Comment (contolini) [inline cfgov/regulations3k/jinja2/regulations3k/landing-page.html:37]:
```suggestion
{% endblock content_main %}
```

File: cfgov/regulations3k/jinja2/regulations3k/search-regulations.html [hunk 6]
```diff
@@ -184,7 +184,7 @@ <h3>Refine results</h3>
 </div>
 {% endif %}
 
-{% endblock content_main %}
+{% endblock content__main %}
```
Comment (contolini) [inline cfgov/regulations3k/jinja2/regulations3k/search-regulations.html:187]:
```suggestion
{% endblock content_main %}
```

File: cfgov/tccp/jinja2/tccp/card.html [hunk 7]
```diff
@@ -952,8 +952,8 @@ <h2 class="h3 u-mb15">
 
 </div>
 
-{% endblock content_main %}
+{% endblock content__main %}
```
Comment (contolini) [inline cfgov/tccp/jinja2/tccp/card.html:955]:
```suggestion
{% endblock content_main %}
```

File: cfgov/tccp/jinja2/tccp/cards.html [hunk 8]
```diff
@@ -59,4 +59,4 @@ <h1>{{ heading }}</h1>
     </button>
 </div>
 
-{% endblock content_main %}
+{% endblock content__main %}
```
Comment (contolini) [inline cfgov/tccp/jinja2/tccp/cards.html:62]:
```suggestion
{% endblock content_main %}
```

File: cfgov/ask_cfpb/jinja2/ask-cfpb/answer-page.html [hunk 9]
```diff
@@ -22,8 +22,8 @@
```
Comment (contolini) [inline cfgov/ask_cfpb/jinja2/ask-cfpb/answer-page.html:22]: The viz checker is showing that Ask pages are a little broken. I believe changing `ask-cfpb-page__answer` to `ask-cfpb-page--answer` on L20 in this file fixes things.

File: cfgov/v1/jinja2/v1/includes/molecules/social-media.html [hunk 10]
```diff
@@ -70,7 +70,7 @@
 <div class="m-social-media
             m-social-media__{{ 'share' if is_share_view else 'follow' }}">
```
Comment (contolini) [inline cfgov/v1/jinja2/v1/includes/molecules/social-media.html:71]:
```suggestion
            m-social-media--{{ 'share' if is_share_view else 'follow' }}">
```

File: cfgov/unprocessed/css/molecules/info-unit.less [hunk 11]
```diff
@@ -43,13 +43,13 @@
...
     margin-bottom: 0;
   }
 
-  .m-info-unit_content,
-  .m-info-unit_heading-text {
+  .m-info-unit__content,
+  .m-info-unit__heading-text {
     margin-left: unit(
       (@img_size + @grid_gutter-width) / @base-font-size-px,
       em
```
Comment (contolini) [inline cfgov/unprocessed/css/molecules/info-unit.less:55]: The print media query on L76 should be changed to `&__image`.

File: cfgov/unprocessed/css/molecules/info-unit.less [hunk 12]
```diff
@@ -29,7 +29,7 @@
   &_heading-link {
```
Comment (contolini) [inline cfgov/unprocessed/css/molecules/info-unit.less:29]: L10 in this file needs `&_image` changed to `&__image`.

**Review by contolini — CHANGES_REQUESTED** [review]
Lookin' good! I think it's just about ready for merge.

File: cfgov/ask_cfpb/jinja2/ask-cfpb/answer-page.html (hunk 9 above)
Comment (anselmbradford) [reply cfgov/ask_cfpb/jinja2/ask-cfpb/answer-page.html:22]: Fixed in https://github.com/cfpb/consumerfinance.gov/pull/8274

File: cfgov/v1/jinja2/v1/includes/molecules/social-media.html (hunk 10 above)
Comment (anselmbradford) [reply cfgov/v1/jinja2/v1/includes/molecules/social-media.html]: Fixed in https://github.com/cfpb/consumerfinance.gov/pull/8274

File: cfgov/unprocessed/css/molecules/info-unit.less (hunk 11 above)
Comment (anselmbradford) [reply cfgov/unprocessed/css/molecules/info-unit.less:55]: Fixed in https://github.com/cfpb/consumerfinance.gov/pull/8274

File: cfgov/unprocessed/css/molecules/info-unit.less (hunk 12 above)
Comment (anselmbradford) [reply cfgov/unprocessed/css/molecules/info-unit.less]: Fixed in https://github.com/cfpb/consumerfinance.gov/pull/8274

**Review by contolini — APPROVED** [review]
🚀 🚀 🚀

**Pull request merged**